src/main/java/org/sonar/plugins/github/GitHubPluginConfiguration.java

```diff
@@ -117,4 +133,52 @@ public boolean tryReportIssuesInline() {
     return !settings.getBoolean(GitHubPlugin.GITHUB_DISABLE_INLINE_COMMENTS);
   }
 
+  /**
+   * Checks if a proxy was passed with command line parameters or configured in the system.
+   * If only an HTTP proxy was configured then it's properties are copied to the HTTPS proxy (like SonarQube configuration)
+   * @return True iff a proxy was configured to be used in the plugin.
+   */
+  public boolean isProxyConnectionEnabled() {
+    if (System.getProperty(HTTP_PROXY_HOSTNAME) != null || System.getProperty(HTTPS_PROXY_HOSTNAME) != null ||
+      System.getProperty(PROXY_SOCKS_HOSTNAME) != null) {
+      return true;
+    }
+    return false;
+  }
+
+  public Proxy getHttpProxy() {
+    try {
+      if (System.getProperty(HTTP_PROXY_HOSTNAME) != null && System.getProperty(HTTPS_PROXY_HOSTNAME) == null) {
+        System.setProperty(HTTPS_PROXY_HOSTNAME, System.getProperty(HTTP_PROXY_HOSTNAME));
+        System.setProperty(HTTPS_PROXY_PORT, System.getProperty(HTTP_PROXY_PORT));
+
+      }
+
+      String proxyUser = System.getProperty(HTTP_PROXY_USER);
+      String proxyPass = System.getProperty(HTTP_PROXY_PASS);
+
+      if (proxyUser != null && proxyPass != null) {
+        Authenticator.setDefault(
+          new Authenticator() {
+            @Override
+            public PasswordAuthentication getPasswordAuthentication() {
+              return new PasswordAuthentication(
+                proxyUser, proxyPass.toCharArray());
+            }
+          });
+      }
+
+      Proxy selectedProxy = ProxySelector.getDefault().select(new URI(endpoint())).get(0);
+
+      if (selectedProxy.type() == Proxy.Type.DIRECT) {
+        LOG.debug("There was no suitable proxy found to connect to GitHub - direct connection is used ");
+      }
+
+      LOG.info("A proxy has been configured - " + selectedProxy.toString());
```

Comment:
It's intentional, since in any case, the connection is configured with proxy - but due to configuration, it might be of a Direct Type. The first message with specific details on an issue will be only displayed on debug mode. There might be use case in which a proxy is not suitable although configured and therefore is not used and proxy.Direct is chosen. The user is informed of this.